docs/release-v0.9.md

```diff
@@ -0,0 +1,379 @@
+
+It is enabled in `config.toml` with the `chain.p2p.private_mode` setting. Whitelisted peers may be set in `config.toml` with `chain.p2p.whitelist_peers`, or managed at runtime with the `kwil-admin whitelist` subcommands.
+This functionality was added in [559b027](https://github.com/kwilteam/kwil-db/commit/559b0279eb10f04a331b4773fd3f0a785a035712).
+
+#### Private RPCs and Authenticated Calls
+
+A "private" RPC mode has been [added](https://github.com/kwilteam/kwil-db/commit/6537114163397b94e4a1aa91513b78a9d45e35c9) with the following features:
+
+* All call requests are authenticated. The Kwil clients handle this automatically when communicating with an RPC provider in private mode. A private key must be configured with the client in order to sign the call requests.
+* Ad hoc SQL queries are prohibited.
+* Transaction status query responses do not include the full transaction, which could potentially contain data that the user is not authorized to access.
+
+For Go SDK users, note that this adds a signature field to the `core/client.CallMessage` type.
+
+Related commits: [6537114](https://github.com/kwilteam/kwil-db/commit/6537114163397b94e4a1aa91513b78a9d45e35c9), [559b027](https://github.com/kwilteam/kwil-db/commit/559b0279eb10f04a331b4773fd3f0a785a035712), [177891a](https://github.com/kwilteam/kwil-db/commit/177891a1498587e133959a10de475271d91b5e96)
+
+### Ability to Rebrand Customized Binary Builds
+
+Projects now have the ability to build custom-branded binaries that group in `kwild`, `kwil-cli`, and `kwil-admin`. These binaries can be given customized default configurations.
+
+The global `cmd.BinaryConfig` structure is added to allow changing the project name and other details.  Projects should set their configuration in an instance of the `cmd/custom.CommonCmdConfig` and then create their application's root command with the `NewCustomCmd` function.
+
+This was added in [2aee07c](https://github.com/kwilteam/kwil-db/commit/2aee07c3c0de5475aa601dd3627463f04c4bb18a) and [efc3af4](https://github.com/kwilteam/kwil-db/commit/efc3af44798b8f3b7373e4500086af9466d7878f).
+
+### Extension Hooks for Genesis Initialization and End-Block Actions
+
+There are now hooks for extension authors to perform arbitrary actions:
+
+* `GenesisHook` is a function that is run exactly once, at network genesis. It can be used to create initial state or perform other setup tasks.
+  Any state changed or error returned should be deterministic, as all nodes will run the same genesis hooks in the same order.
+  Named genesis hooks are registered with the `extensions/hooks.RegisterGenesisHook` function.
+* `EndBlockHook` is a function that is run at the end of each block, after all of the transactions in the block have been processed, but before the any state has been committed.
+  It is meant to be used to alter state, send data to external services, or perform cleanup tasks for other extensions.
```

Comment:
No, only the state committed to the DB needs to be deterministic. If you did something like log a message, send some metrics, or otherwise that is not deterministic, that is ok, as long as it does not affect your processing of the block

Reply:
I see what both of you are saying here.  But perhaps should there be a suggestion that if the end-block hook performs some action off-chain then it should be idempotent, or does this entirely depend on what it does with said external services?  If there's any guidance to add, please let me know